Bottom: f09f8b9d49fe5451ec3d2c4b634cabca0edfbd36
Top:    63da2c89aa17a635abaebc8a8de2936a1496418a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 15:55:47 +0100

hid/gtk: Replace GTK_WIDGET_REALIZED() test with a check for window != NULL

This avoids the use of the deprecated GTK_WIDGET_REALIZED macro, without
needing to use the GTK2.20 function gtk_widget_get_realized(). The check
for a NULL window should be sufficient here I hope.


---

diff --git a/src/hid/gtk/gui-pinout-preview.c b/src/hid/gtk/gui-pinout-preview.c
index c4bef20..79fc0f2 100644
--- a/src/hid/gtk/gui-pinout-preview.c
+++ b/src/hid/gtk/gui-pinout-preview.c
@@ -191,14 +191,14 @@ ghid_pinout_preview_set_property (GObject * object, guint property_id,
 				  const GValue * value, GParamSpec * pspec)
 {
   GhidPinoutPreview *pinout = GHID_PINOUT_PREVIEW (object);
+  GdkWindow *window = gtk_widget_get_window (GTK_WIDGET (pinout));
 
   switch (property_id)
     {
     case PROP_ELEMENT_DATA:
       pinout_set_data (pinout, (ElementType *)g_value_get_pointer (value));
-      if (GTK_WIDGET_REALIZED (GTK_WIDGET (pinout)))
-        gdk_window_invalidate_rect (
-            gtk_widget_get_window (GTK_WIDGET (pinout)), NULL, FALSE);
+      if (window != NULL)
+        gdk_window_invalidate_rect (window, NULL, FALSE);
       break;
     default:
       G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
